continuous_integration/docker/hadoop/files/etc/hadoop/conf.kerberos/yarn-site.xml

```diff
@@ -61,18 +67,19 @@
     </property>
 
     <property>
-        <name>yarn.nodemanager.resource.cpu-vcores</name>
-        <value>16</value>
+        <name>yarn.nodemanager.resource.memory-mb</name>
+        <value>4096</value>
```

Comment:
You probably know, but YARN doesn't actually respect these as limits, it's only used for internal bookkeeping; so it doesn't matter is the worker actually had this much memory available.

Reply:
Ooh, that is critical knowledge. I wanted to stay within memory limits and constrain this to avoid crashing things during tests. I had seen a warning about possible "thrashing" of workers or similar.

Ideas on how to make us limit ourselves better?

Reply:
Is this config also pointless?

```xml
    <property>
        <name>yarn.scheduler.minimum-allocation-mb</name>
        <value>32</value>
    </property>

    <property>
        <name>yarn.resource-types.memory-mb.increment-allocation</name>
        <value>${yarn.scheduler.minimum-allocation-mb}</value>
    </property>
```

Reply:
I did get tripped up by this once, where where the default total memory resource of the cluster was 4096 and the minimum allocation was 1024, so you would get no more than 4 containers no matter what the actual memory requirements of requests were. The default minimum still seems to be 1024 https://hadoop.apache.org/docs/r3.0.1/hadoop-yarn/hadoop-yarn-common/yarn-default.xml 

I would make these numbers small, or else just wait in case something ever goes wrong.

Reply:
@martindurant I found that in the test_yarn_backend.py file, the YarnBackend is configured with memory for the workers. I lowered it from 512 to 128 and things may have improved, not sure yet - intermittency =/